packages/insomnia-app/app/network/network.js

```diff
@@ -109,13 +109,17 @@ const LIBCURL_DEBUG_MIGRATION_MAP = {
   '': '',
 };
 
-let cancelRequestFunction = null;
+const cancelRequestFunctionMap = {};
 let lastUserInteraction = Date.now();
 
-export async function cancelCurrentRequest() {
-  if (typeof cancelRequestFunction === 'function') {
-    return cancelRequestFunction();
+export async function cancelRequestById(requestId) {
+  if (cancelRequestFunctionMap.hasOwnProperty(requestId)) {
+    const cancelRequestFunction = cancelRequestFunctionMap[requestId];
+    if (typeof cancelRequestFunction === 'function') {
+      return cancelRequestFunction();
+    }
   }
+  console.log(`WARN: cancelFunction for ${requestId} not found.`);
```

Comment:
Small suggestion to update the error message here, nice observation to add it in 😄 

```suggestion
  console.log(`[network] Failed to cancel req=${requestId} because cancel function not found`);
```